src/eventHandlers/ResourceBookingEventHandler.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Handle events for ResourceBooking.
+ */
+
+const { Op } = require('sequelize')
+const models = require('../models')
+const logger = require('../common/logger')
+const helper = require('../common/helper')
+const JobService = require('../services/JobService')
+const JobCandidateService = require('../services/JobCandidateService')
+
+/**
+ * When ResourceBooking's status is changed to `assigned`
+ * the corresponding JobCandidate record (with the same userId and jobId)
+ * should be updated with status `selected`
+ *
+ * @param {String} jobId the job id
+ * @param {String} userId the user id
+ * @returns {undefined}
+ */
+async function selectJobCandidate (jobId, userId) {
+  const candidates = await models.JobCandidate.findAll({
+    where: {
+      jobId,
+      userId,
+      status: {
+        [Op.not]: 'selected'
+      },
+      deletedAt: null
+    }
+  })
+  await Promise.all(candidates.map(candidate => JobCandidateService.partiallyUpdateJobCandidate(
+    helper.getAuditM2Muser(),
+    candidate.id,
+    { status: 'selected' }
+  ).then(result => {
+    logger.info({
+      component: 'ResourceBookingEventHandler',
+      context: 'selectJobCandidate',
+      message: `id: ${result.id} candidate got selected.`
+    })
+  })))
+}
+
+/**
+ * Update the status of the Job to assigned when it positions requirement is fullfilled.
+ *
+ * @param {Object} job the job data
+ * @returns {undefined}
+ */
+async function assignJob (job) {
+  if (job.status === 'assigned') {
+    logger.info({
+      component: 'ResourceBookingEventHandler',
+      context: 'assignJob',
+      message: `job with projectId ${job.projectId} is already assigned`
+    })
+    return
+  }
+  const resourceBookings = await models.ResourceBooking.findAll({
+    where: {
+      status: 'assigned',
+      deletedAt: null
+    }
+  })
```

Comment:
@imcaizheng I'm not sure how this work. By logic, we have to count only `ResourceBooking` associated with a particular Job only. And if a Job has ResourceBooking associated the same quantity as `numPositions` then mark Job as assigned. This logic seems to count all the ResourceBooking in the DB.

Reply:
Thanks for pointing it out. I will fix it by adding filter of `jobId`. Again, it confuses me if `jobId` is an optional field for the ResourceBooking model.

Reply:
@imcaizheng let' keep `jobId` optional until I confirm with PM. And here if ResourceBooking doesn't have `jobId` then don't run the automatic update for Job as a resource doesn't have associated job.